server/src/test/java/org/elasticsearch/indices/SystemIndexDescriptorTests.java

```diff
@@ -34,36 +34,48 @@ public class SystemIndexDescriptorTests extends ESTestCase {
      */
     public void testValidation() {
         {
-            Exception ex = expectThrows(NullPointerException.class, () -> new SystemIndexDescriptor(null, randomAlphaOfLength(5)));
+            Exception ex = expectThrows(
+                NullPointerException.class,
+                () -> SystemIndexDescriptorUtils.createUnmanaged(null, randomAlphaOfLength(5))
+            );
             assertThat(ex.getMessage(), containsString("must not be null"));
         }
 
         {
-            Exception ex = expectThrows(IllegalArgumentException.class, () -> new SystemIndexDescriptor("", randomAlphaOfLength(5)));
+            Exception ex = expectThrows(
+                IllegalArgumentException.class,
+                () -> SystemIndexDescriptorUtils.createUnmanaged("", randomAlphaOfLength(5))
+            );
             assertThat(ex.getMessage(), containsString("must at least 2 characters in length"));
         }
 
         {
-            Exception ex = expectThrows(IllegalArgumentException.class, () -> new SystemIndexDescriptor(".", randomAlphaOfLength(5)));
+            Exception ex = expectThrows(
+                IllegalArgumentException.class,
+                () -> SystemIndexDescriptorUtils.createUnmanaged(".", randomAlphaOfLength(5))
```

Comment:
It seems odd to be asserting behavior of a test utils method (at least for what seems to be real production rules). Is there another ctor that could be called here to check the same validation is done, but assures that it is the production code doing the validation, not the test helper method? (ditto for other calls to createUnmanaged in this file)